cli/src/ai/render-markdown.ts

```
@@ -0,0 +1,85 @@
+    if (raw.trimStart().startsWith('```')) {
+      inCodeBlock = !inCodeBlock
+      out.push(stylize(ANSI.gray, raw))
+      continue
+    }
+    if (inCodeBlock) {
+      out.push(stylize(ANSI.cyan, raw))
+      continue
+    }
+
+    // Headers
+    const headerMatch = raw.match(/^(#{1,6})\s+(.+)$/)
```

ESLint

[error] The quantifier '\s+' can exchange characters with '.+'. Using any string accepted by /[\t\x0b\f \xa0\u1680\u2000-\u200a\u202f\u205f\u3000\ufeff]+/, this can be exploited to cause at least polynomial backtracking.

(regexp/no-super-linear-backtracking)